COH_Australia/Australia_importer.py

```diff
@@ -0,0 +1,578 @@
+EDIT_SUMMARY = 'importing Australia #COH #WLM #au'
+
+
+class ImporterBot(object):
+    """Bot to enrich/create info on Wikidata for Australian heritage items."""
+
+    def __init__(self, base_path, new=False, cutoff=None, preview_file=None):
+        """
+        Initialise the ImporterBot.
+
+        :param base_path: path to the output directory
+        :param new: whether to also create new items
+        :param cutoff: the number of items to process before stopping. None
+            being interpreted as all.
+        :param preview_file: run in demo mode (create previews rather than
+            live edits) and output the result to this file.
+        """
+        self.repo = pywikibot.Site().data_repository()
+        self.wd = WdS(self.repo, EDIT_SUMMARY)
+        self.new = new
+        self.cutoff = cutoff
+        self.preview_file = path.join(base_path, preview_file)
+        if preview_file:
+            self.demo = True
+        else:
+            self.demo = False
+        self.preview_data = []
+
+        self.set_references()
+        self.place_id_p = 'P3008'  # unique identifier property
+        self.country = self.wd.QtoItemPage('Q408')
+        self.states = self.make_states_map()
+        self.hectares = self.wd.QtoItemPage(helpers.get_unit_q('ha'))
+        self.make_status_and_instance_map()
+
+        self.place_id_items = helpers.fill_cache_wdqs(
+            self.place_id_p, no_strip=True)
+
+    def set_references(self):
+        """Set the three types of references needed."""
+        pywikibot.warning('These links and dates should be updated')
+        self.ref = {
+            'national': self.make_url_ref(
+                'http://national.foo.com',
+                '2099-01-01'),
+            'commonwealth': self.make_url_ref(
+                'http://commonwealth.foo.com',
+                '2099-01-01')
+        }
+        self.coord_ref = {
+            'national': self.make_url_ref(
+                'http://www.environment.gov.au/heritage/places/national-heritage-list',  # noqa
+                '2017-08-13'),
+            'commonwealth': self.make_url_ref(
+                'https://data.gov.au/dataset/57720684-4948-45db-a2c8-37259d531d87',  # noqa
+                '2017-08-13',
+                '2017-07-10')
+        }
+
+    def make_status_and_instance_map(self):
+        """Construct mapping for cultural heritage status and instance type."""
+        self.status = {
+            'national': self.wd.QtoItemPage('Q20747146'),
+            'commonwealth': self.wd.QtoItemPage('Q30108476')
+        }
+        self.instance_type = {
+            'indigenous': self.wd.QtoItemPage('Q38048771'),
+            'historic': self.wd.QtoItemPage('Q38048707'),
+            'natural': self.wd.QtoItemPage('Q38048753')
+        }
+
+    def make_states_map(self):
+        """
+        Retrieve the state/territory mappings from Wikidata.
+
+        Also tries to makes up stuff for the EXT and OS codes.
```

Comment:
"tries to makes up stuff" -> "tries to match items"?

Reply:
What does "make up stuff"  not sounds serious ;)
Done

Reply:
I think it implies inventing something out of thin air... But what we're doing here is straightforward matching.